docs/API.md

```diff
@@ -2129,6 +2129,129 @@ if err != nil {
 fmt.Printf("%+v\n", versioningConfig)
 ```
 
+<a name="SetBucketReplication"></a>
+
+### SetBucketReplication(ctx context.Context, bucketname, cfg replication.Config, replicationARN string) error
+Set replication configuration on a bucket. Prior to setting replication configuration, replication endpoint should have been defined on MinIO
+using `mc admin bucket replication set` to associate the bucket `bucketName` with the replication endpoint. Next, this call requires a replication ARN which can be fetched from MinIO server using `mc admin bucket remote`
+
+__Parameters__
+
+|Param   |Type   |Description   |
+|:---|:---| :---|
+|`ctx`  | _context.Context_  | Custom context for timeout/cancellation of the call|
+|`bucketName` | _string_  |Name of the bucket|
+|`cfg` | _replication.Config_  |Replication configuration to be set |
+|`replicationARN` | _string_  |Replication ARN from MinIO server identifying the replication endpoint |
```

Comment:
I think personally we should extend the

```
<ReplicationConfiguration>
   <Role>string</Role>
   <ReplicationArn></ReplicationArn> --> optional for AWS
   <Rule>
      <DeleteMarkerReplication>
         <Status>Disabled</Status>
      </DeleteMarkerReplication>
      <Destination>
         <Bucket>string</Bucket>
         <StorageClass>string</StorageClass>
      </Destination>
      <Filter>
         <And>
            <Prefix>string</Prefix>
            <Tag>
               <Key>string</Key>
               <Value>string</Value>
            </Tag>
            ...
         </And>
         <Prefix>string</Prefix>
         <Tag>
            <Key>string</Key>
            <Value>string</Value>
         </Tag>
      </Filter>
      <ID>string</ID>
      <Prefix>string</Prefix>
      <Priority>integer</Priority>
      <Status>string</Status>
   </Rule>
</ReplicationConfiguration>
```

Reply:
this is a good idea, makes it cleaner. Done